### PR TITLE
Implement and enhance UI

### LINKED ISSUE #5: Enhance UI
Add better UI to the project, probably use a ReactJS framework such as Ant Design or Material UI

### PR DESCRIPTION
**Pull request by paramsiddharth**
# Related Issue
- Closes #5: **Enhance UI**

# Proposed Changes
- Separation of concerns: Move business logic, components, and controls out of `App.js` and import them as needed.
- Create GitHub-like UI using Reactstrap, a Bootstrap extension library made specifically for React. The UI is responsive i. e. Is optimized for all screen-sizes.
- Scaffold a component for visualization and bind API data. This would allow other participants to start working on the visualizations.

# Additional Info
- The UI inside `<App/>` is distributed into 3 parts, `<Header/>`, `<Main/>`, and `<Footer/>`.
- The GitHub API handling logic is now in `services/GitHub.js`.
- `<Visualization data={info} />` has `props.data` bound to the output of API calls done by `GitHub.js`. `console.log(data);` may simply be removed in future versions.

# Checklist
- [ ] Tests
- [ ] Translations
- [ ] Documentation

# Screenshots

Original              |  Updated
:--------------------------:|:-----------------------:
image | image

### CODE REVIEW

File: src/App.test.js [hunk 1]
```diff
@@ -3,6 +3,6 @@ import App from './App';
 
 test('renders learn react link', () => {
   render(<App />);
-  const linkElement = screen.getByText(/Enter your github username/i);
+  const linkElement = screen.getByText(/Username:\S?/i);
```
Comment (adityabisoi) [inline src/App.test.js]: please remove the test for now, we'll cover it in another PR.

File: src/components/header.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { Jumbotron } from 'reactstrap';
+import './header.css';
+
+const Header = () => {
+	return (
+		<header >
+			<Jumbotron className='m-0 bg-dark text-white text-left rounded-0'>
+				<a href='./'><h1 className='display-3 font-weight-normal'>GitHub API Playground</h1></a>
+				<p className='d-none d-md-block lead font-weight-normal'>A playground for GitHub's REST API.</p>
```
Comment (adityabisoi) [inline src/components/header.js]: Please remove the sub-text for now.

File: src/components/header.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { Jumbotron } from 'reactstrap';
+import './header.css';
+
+const Header = () => {
+	return (
+		<header >
+			<Jumbotron className='m-0 bg-dark text-white text-left rounded-0'>
+				<a href='./'><h1 className='display-3 font-weight-normal'>GitHub API Playground</h1></a>
```
Comment (adityabisoi) [inline src/components/header.js]: Make the header smaller like the usual-sized navbars on websites. Also, please change the background to a better color.

**Review by adityabisoi — CHANGES_REQUESTED** [review]

**Comment by adityabisoi** [comment]
@paramsiddharth It's always a better option to create a new branch for each issue you resolve. This keeps workflow clean.

**Comment by paramsiddharth** [comment]
> @paramsiddharth It's always a better option to create a new branch for each issue you resolve. This keeps workflow clean.

Noted! I'll take care of that next time.

**Pull request merged**